.github/workflows/check_new_kernel_configs.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+
+def get_data_from_config(input_file):
+    with open(input_file, 'r') as file:
+        input_config_data = file.read()
+    return input_config_data
+
+def check_kernel(input_file):
+    match = re.search(r'SPECS/(.*?)/', input_file)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+def check_config_arch(input_config_data):
+    if "Linux/x86_64" in input_config_data:
+        return "AMD64"
+    elif "Linux/arm64" in input_config_data:
+        return "ARM64"
+    else:
+        return None
+
+# find the lines in the diff that contain +/- CONFIG
+# ignore the lines that contain @
+# return the set of words that follow +/-
+def find_matching_lines(input_string):
+    pattern = r'(?!.*@)[+-]\s*.*CONFIG.*'
+    matching_lines = re.findall(pattern, input_string)
+    holder=[re.sub(r"\+|\-|=y|=m|\#|is not set", r"", s).strip() for s in matching_lines]
+    config_set = set(holder)
+    return config_set
+
+# parse diff for new kernel configs
+# check if they are in required configs
```

Comment:
Nit-pick: let's be consistent with how we write comments. My personal preference: begin with a capital letter and use proper punctuation. To the best of our abilities that is.;)